### DISCUSSION TITLE
What to expect for Hankaku & Zenkaku?

### DISCUSSION THREAD

**Discussion by maxl2287** [camaraproject/KnowYourCustomer#122] [Q&A] [answered]
Hi,

sorry I am not so familiar with the japanese language, but I want to understand how these name-forms are working with the API.
Imagine we are using fill-in and get back some "name" from the southbound-API.

## Questions
- In which cases does it make sense to return `nameKanaHankaku` & `nameKanaZenkaku` ?
- Is it usual to write also "Latin"-names, like "Lilly", "Max", in zenkaku then?

`
Max -> Ｍａｘ

Lilly -> Ｌｉｌｌｙ
`
- For which names does it make sense to write them in zenkaku / hankaku?

Thanks in advance!
Max 🚀

**Comment by ToshiWakayama-KDDI** [comment, marked as answer]
Hi @maxl2287 ,

Thank you for the question.  Good point; actually unusual, and they are Japanese language specific character sets, even though they are part of UTF-8 Unicode.  For us Japanese, we use Chinese characters (Japanese version) for names, and in addition, we use Kana names, which correspond to pronounciation of the names and which are written in Kana Zenkaku and Kana Hankaku char set.
 
For example, my first name is Toshiyasu, which is composed of four syllables and can be writen as 'ト シ ヤ ス' in Kana Zenkaku, and as 'ﾄ ｼ ﾔ ｽ' in Kana Hankaku.  Hope you can see the differences.  And, why 'federca' and 'ｆｅｄｅｒｉｃａ' are used as the API examples is that we think names written in Kana Zenkaku and Kana Hankaku do not mean anything for international API users who do not understand Japanese language.
 
Unusual, but It may still be possible to use Kana Zenkaku and Kana Hankaku for Latin-names, e.g 'フェデリカ'/'ﾌｪﾃﾞﾘｶ' for Federica, 'マックス'/'ﾏｯｸｽ' for Max, 'リリー'/'ﾘﾘｰ' for Lilly, just if a Service Provider should want, by writing pronounciation of these Latin-names using Kana Zenkaku/ Kana Hankakku. 
 
Best regards,
Toshi
KDDI

**Reply by HuubAppelboom** [reply]
Indeed, 
Best regards
ハーブ・アッペルブーム

**Reply by maxl2287** [reply]
Thanks for this detailed explaination! 
Unfortunately joining Japanese lessons were canceled, when I was at school 10 years ago. 😅 

本当にありがとう 😄
